src/snowflake/snowpark/_internal/compiler/large_query_breakdown.py

```diff
@@ -233,9 +242,23 @@ def _find_node_to_breakdown(self, root: TreeNode) -> Optional[TreeNode]:
                     else:
                         # don't traverse subtrees if parent is a valid candidate
                         next_level.append(child)
+                        self.breakdown_failure_summary[reason_if_invalid.value] += 1
+                        if (
+                            reason_if_invalid
+                            == CompilationStageTelemetryField.INVALID_DUE_TO_EXTERNAL_CTE_REF
+                        ):
+                            invalid_due_to_external_cte_ref_encountered = True
 
             current_level = next_level
 
+        if candidate_node is None:
+            self.breakdown_failure_summary[
```

Comment:
instead of one summary, maybe we can get a list of summary for each logical plan each one includes 1) num_partitions_made 2) invalid nodes with reason
with those should be good enough for us to debug

Reply:
updated